include/pybind11/detail/smart_holder_type_casters.h

```diff
@@ -106,6 +115,27 @@ class modified_type_caster_generic_load_impl {
         return false;
     }
 
+    bool try_as_void_ptr_capsule(handle src) {
+        std::string type_name = cpptype->name();
+        detail::clean_type_id(type_name);
+
+        // Convert `a::b::c` to `a_b_c`
+        replace_all(type_name, "::", '_');
+
+        std::string as_void_ptr_function_name = "as_" + type_name;
```

Comment:
Very tiny nit, this could be an allocation of "as_" and += type_name and remove an allocation.